README.md

```diff
@@ -177,83 +181,95 @@ Now each of the above list must be converted to bytearrays, i.e. `Uint8Array` an
   // Deterministically generated params
-  const paramsDeterministc = SignatureParamsG1.generate(messageCount, label);
-  
+  const paramsDeterministc = BBSSignatureParams.generate(messageCount, label);
+
   // Deterministic params can be extended if messageCount changes, say to 5 or 3
-  const paramsNew = paramsDeterministc.adapt(5);
-  const paramsNeww = paramsDeterministc.adapt(3);
+  const paramsDeterministc5 = paramsDeterministc.adapt(5);
+  const paramsDeterministc3 = paramsDeterministc.adapt(3);
   ```
   
   Generating a keypair once signature parameters are created.
 
   ```ts
-  const keypair1 = BBSPlusKeypairG2.generate(paramsDeterministc);
+  import { BBSKeypair } from '@docknetwork/crypto-wasm-ts';
+  const keypair = BBSKeypair.generate(paramsDeterministc);
   const sk = keypair.secretKey;
   const pk = keypair.publicKey;
   ```
 
+  #### ByteArray messages
+  Each one of the messages should be a Uint8Array 
+
+  ```ts
+  const messages: Uint8Array[] = ["NYC", "John", "Smith", "12345678"].map(element => stringToBytes(element));
+  ```
+
 #### Signing and verification
 
 When the messages are arbitrary bytes, they need to be encoded to a field element (a number in certain range). You can either let the signing function encode it by passing 
 the `encode` argument as true to encode it using your own encoding function.
   
   Letting the signing function encode  
   ```ts
-  // messages is a list of bytearrays and converted as mentioned above  
-  const messages: Uint8Array[] = [...];
+  import { BBSSignature } from '@docknetwork/crypto-wasm-ts';
   
   // The signing function will encode bytes to a field element as true is passed
-  const sig = SignatureG1.generate(messages, sk, params, true);
+  const sig = BBSSignature.generate(messages, sk, paramsDeterministc, true);
   
   // As the messages are not encoded, pass true to the verification function to make it encode messages before verifying the signature.
-  const result = sig.verify(messages, pk, params, true);
-  expect(result.verified).toEqual(true);
+  const result = sig.verify(messages, pk, paramsDeterministc, true);
+  
+  result.verified // true
   ```
   
   Passing pre-encoded messages to signing function
   ```ts
-  // messages is a list of bytearrays and converted as mentioned above  
-  const messages: Uint8Array[] = [...];
-  
+  const encodedMessages = [];
+
   for (let i = 0; i < messages.length; i++) {
     encodedMessages.push(generateFieldElementFromBytes(messages[i]));
   }
   // The signing function will not encode as false is passed
-  const sig = SignatureG1.generate(encodedMessages, sk, params, false);
+  const sig = BBSSignature.generate(encodedMessages, sk, params, false);
 
   // As the messages are pre-encoded, pass false to the verification function to avoid encoding messages before verifying the signature.
   const result = sig.verify(encodedMessages, pk, params, false);
-  expect(result.verified).toEqual(true);
+  result.verified // true
```

Comment:
Either keep the `expect`, or do something like:

```
if (!result.verified){
  throw new Error("Signature verification failed");
}
```

Reply:
Even I prefer keeping an "error throwing" mechanism over a comment even for illustration purposes. Wdyt about `console.assert` over `expect`? The former is available  in browsers and node.js

Reply:
I didn't know of `console.assert` - I like it. Yes, let's replace the `throw`/`expect` with `console.assert`.

Reply:
Sounds good! I used `console.assert(..);`